lib/charms/operator_libs_linux/v0/juju_systemd_notices.py

```diff
@@ -310,8 +346,10 @@ async def _send_juju_notification(service: str, state: str) -> None:
     if service.endswith(".service"):
         service = service[0:-len(".service")]  # fmt: skip
 
+    global _observed_services
```

Comment:
You actually only need the `global` statement when you're re-assigning the global; here you're just referencing it (you've already assigned the name at the global level), so no need for `global`.

I'd prefer if this was more explicitly passed down, rather than a global, in any case. Is that relatively easy to achieve? (Similar with `_juju_unit`.)

Reply:
`global` can be used for explicitly declaring that we're using a global variable within the scope of the function, but yes, we can reference it implicitly as well.

I prefer to use the globals here since they're effectively constants that we're setting in `_main()` before starting the DBus observer. I don't necessarily want to expand the number of arguments that the daemon's methods must accept to pass down static values. I also think it's simpler as we know that all references to `_juju_unit`, `_observed_services`, and `_service_states` are referring to the same variable rather than needing to chase down references within the daemon's functions and see what is bubbled back up.